### PR TITLE
Add Test for Mtom Stream Upload 

### PR DESCRIPTION
**Pull request by afifi-ins**

### CODE REVIEW

**Comment by mconnew** [comment]
Can you remove the commented out CoreWCF service and fix the global.json conflict

**Comment by afifi-ins** [comment]
> Can you remove the commented out CoreWCF service and fix the global.json conflict

Thanks, Matt, for your suggestions. There is a problem when using the Mtom headers method, the client will not return the correct number of uploaded bytes, but it will always return zero (seems like the client callbacks are not called). Calling Assert() for test validation will always fail. 

```c#
long result = serviceProxy.UploadStream(new BloatedStream(uploadBytes));
Assert.Equal(uploadBytes, result); //result=0
```

**Comment by mconnew** [comment]
In working on some other fixes, I actually discovered an async bug in the mtom message encoder when running on a concurrency constrained SynchronizationContext. That fix has been merged so if you rebase, I think the CoreWCF based service will work now.

File: src/System.Private.ServiceModel/tests/Scenarios/Binding/Http/BasicHttpBindingTests.4.0.0.cs [hunk 1]
```diff
@@ -359,4 +362,43 @@ public static void DecompressionEnabled_Echo_RoundTrips_String(bool decompressio
...
+    {
+        ChannelFactory<MtomBindingTestHelper.IMtomStreamingService> factory = null;
+        MtomBindingTestHelper.IMtomStreamingService serviceProxy = null;
+        WebApplication app = null;
+        try
+        {
+            // *** SETUP *** \\
+            // CoreWCF Server
+            app = MtomBindingTestHelper.BuildCoreWCFService();
+            app.Start();
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Scenarios/Binding/Http/BasicHttpBindingTests.4.0.0.cs:382]: The app needs to be stopped, and Dispose'd.

**Comment by mconnew** [comment]
I think your branch might need to be rebased on main. I believe that's what's causing the failures.

**Pull request merged**